2009-10-29  Ulrich Drepper  <drepper@redhat.com>

	[BZ #10847]
	* sysdeps/gnu/getutmp.c: Allow compatibility code to play around with
	getutmpx symbol.
 
--- a/sysdeps/gnu/getutmp.c
+++ b/sysdeps/gnu/getutmp.c
@@ -19,10 +19,12 @@
 #include <assert.h>
 #include <string.h>
 #include <utmp.h>
+#ifndef _UTMPX_H
 /* This is an ugly hack but we must not see the getutmpx declaration.  */
-#define getutmpx XXXgetutmpx
-#include <utmpx.h>
-#undef getutmpx
+# define getutmpx XXXgetutmpx
+# include <utmpx.h>
+# undef getutmpx
+#endif
 
 void
 getutmp (const struct utmpx *utmpx, struct utmp *utmp)
